prusti/src/ide_helper/compiler_info.rs

```diff
@@ -0,0 +1,128 @@
+use super::{call_finder, query_signature};
+use prusti_interface::{environment::Environment, specs::typed};
+use prusti_rustc_interface::{
+    hir::def_id::DefId,
+    span::{source_map::SourceMap, Span},
+};
+use prusti_viper::ide::vsc_span::VscSpan;
+use serde::{ser::SerializeStruct, Serialize};
+
+/// This struct will be passed to prusti-assistant containing information
+/// about the program that is currently being verified
+#[derive(Serialize)]
+pub struct IdeInfo {
+    procedure_defs: Vec<ProcDef>,
+    function_calls: Vec<ProcDef>,
+    queried_source: Option<String>,
+    // additionally this will contain:
+    // function_calls:
+    // ... we'll see
+}
+
+impl IdeInfo {
+    pub fn collect(
+        env: &Environment<'_>,
+        procedures: &Vec<DefId>,
+        def_spec: &typed::DefSpecificationMap,
+    ) -> Self {
+        let procs = collect_procedures(env, procedures, def_spec);
+        let source_map = env.tcx().sess.source_map();
+        let fncalls: Vec<ProcDef> = collect_fncalls(env)
+            .into_iter()
+            .map(|(name, defid, sp)| ProcDef {
+                name,
+                defid,
+                span: VscSpan::from_span(&sp, source_map).unwrap(),
+            })
+            .collect();
```

Comment:
```suggestion
            .collect::<Vec<_>>();
```
(bit nicer than explicitly annotating `fncalls` with a type hint)